[16.0][MIG] ddmrp

DavidJForgeFlow opened the pull request:
Migration to 16.0


@ForgeFlow

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: ddmrp-11.0/ddmrp-11.0-ddmrp
Translate-URL: https://translation.odoo-community.org/projects/ddmrp-11-0/ddmrp-11-0-ddmrp/

LoisRForgeFlow committed:
[11.0][FIX] ddmrp: ADU past looking was considering one extra day
because todays date is considered in the domain.

OCA-git-bot committed:
[UPD] README.rst

sergiotm-aserti committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (170 of 170 strings)

Translation: ddmrp-11.0/ddmrp-11.0-ddmrp
Translate-URL: https://translation.odoo-community.org/projects/ddmrp-11-0/ddmrp-11-0-ddmrp/es/

LoisRForgeFlow committed:
[13.0][MIG/REF] ddmrp:
* new dedicated model to stock buffer.
* general UX improvements
* Eficent rebranded to ForgeFlow.
* Relicenced to LGPL as agreed by the authors.

LoisRForgeFlow committed:
[IMP] ddmrp: black, isort, prettier

oca-travis committed:
[UPD] Update ddmrp.pot

OCA-git-bot committed:
[UPD] README.rst

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: ddmrp-13.0/ddmrp-13.0-ddmrp
Translate-URL: https://translation.odoo-community.org/projects/ddmrp-13-0/ddmrp-13-0-ddmrp/

LoisRForgeFlow committed:
[13.0][IMP] ddmrp: update icon and add dedicated settings block.

LoisRForgeFlow committed:
[13.0][IMP] ddmrp: add `ddmrp_auto_update_nfp` option.

oca-travis committed:
[UPD] Update ddmrp.pot

OCA-git-bot committed:
[UPD] changelog

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
ddmrp 13.0.1.1.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: ddmrp-13.0/ddmrp-13.0-ddmrp
Translate-URL: https://translation.odoo-community.org/projects/ddmrp-13-0/ddmrp-13-0-ddmrp/

LoisRForgeFlow committed:
[13.0][UPD] ddmrp: add ddmrp_adjustment to setting wizard

oca-travis committed:
[UPD] Update ddmrp.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: ddmrp-13.0/ddmrp-13.0-ddmrp
Translate-URL: https://translation.odoo-community.org/projects/ddmrp-13-0/ddmrp-13-0-ddmrp/

LoisRForgeFlow committed:
[13.0][UPD] ddmrp: add latest modules to settings panel

oca-travis committed:
[UPD] Update ddmrp.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: ddmrp-13.0/ddmrp-13.0-ddmrp
Translate-URL: https://translation.odoo-community.org/projects/ddmrp-13-0/ddmrp-13-0-ddmrp/

didierdonze committed:
Translated using Weblate (French)

Currently translated at 6.1% (15 of 247 strings)

Translation: ddmrp-13.0/ddmrp-13.0-ddmrp
Translate-URL: https://translation.odoo-community.org/projects/ddmrp-13-0/ddmrp-13-0-ddmrp/fr/

didierdonze committed:
Translated using Weblate (French)

Currently translated at 15.0% (37 of 247 strings)

Translation: ddmrp-13.0/ddmrp-13.0-ddmrp
Translate-URL: https://translation.odoo-community.org/projects/ddmrp-13-0/ddmrp-13-0-ddmrp/fr/

JordiBForgeFlow committed:
add method refresh_button in stock.buffer to be called
from the buffer views instead of cron_actions

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: ddmrp-13.0/ddmrp-13.0-ddmrp
Translate-URL: https://translation.odoo-community.org/projects/ddmrp-13-0/ddmrp-13-0-ddmrp/

LoisRForgeFlow committed:
[13.0][IMP] ddmrp: add option to auto procure buffers.

AdriaGForgeFlow committed:
[13.0][FIX] ddmrp: quantity on hand in ddmrp buffer chart

JordiBForgeFlow committed:
use split_every in buffer crons in order to allow more frequent
commits and locking the buffers for too long.

JordiBForgeFlow committed:
commit only if we are not testing

LoisRForgeFlow committed:
[13.0][FIX] ddmrp: quantity on hand in ddmrp buffer chart (continuation)

oca-travis committed:
[UPD] Update ddmrp.pot

OCA-git-bot committed:
ddmrp 13.0.1.2.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: ddmrp-13.0/ddmrp-13.0-ddmrp
Translate-URL: https://translation.odoo-community.org/projects/ddmrp-13-0/ddmrp-13-0-ddmrp/

OCA-git-bot committed:
ddmrp 13.0.1.2.1

Commit:
[FIX] ddmrp: Fix error when adding a purchase order line

To reproduce:

* Open a Purchase Order with at least one line created by a DDMRP buffer
* Add a new line

When selecting the product, we'll get:

    File "/odoo/src/odoo/api.py", line 374, in _call_kw_multi
      result = method(recs, *args, **kwargs)
    File "/odoo/src/odoo/models.py", line 6074, in onchange
      snapshot0.fetch(name)

    [...redacted...]

    File "/odoo/src/odoo/models.py", line 3919, in _compute_field_value
      getattr(self, field.compute)()
    File "/odoo/external-src/ddmrp/stock_buffer_capacity_limit/models/stock_buffer.py", line 20, in _compute_procure_recommended_qty
      res = super()._compute_procure_recommended_qty()
    File "/odoo/external-src/ddmrp/ddmrp/models/stock_buffer.py", line 432, in _compute_procure_recommended_qty
      subtract_qty = self.sudo()._quantity_in_progress()
    File "/odoo/external-src/ddmrp/ddmrp/models/stock_buffer.py", line 157, in _quantity_in_progress
      res[buffer.id] += poline.product_uom._compute_quantity(
    KeyError: 3

The problem is that, in this context, the field is computed by an
onchange, which wraps the records with a NewID record. The original
record is stored in "_origin".

The idiom:

  {}.fromkeys(self.ids, 0.0)

Will automatically unwrap the ids, so we prevent comparing a "<NewID
origin=1>" with the actual ID, which is therefore not found.

In _compute_procure_recommended_qty, use _origin if available or skip
the computation.

LoisRForgeFlow committed:
[13.0][FIX] ddmrp: when auto updating nfp, action can be triggered
by users without write access on buffers, thus we do it with sudo.

OCA-git-bot committed:
ddmrp 13.0.1.2.2

LoisRForgeFlow committed:
[13.0][FIX] ddmrp: correct computation of purchased dlt when there are different lead times for each variant.

OCA-git-bot committed:
ddmrp 13.0.1.2.3

LoisRForgeFlow committed:
[13.0][IMP] ddmrp: automatically assign new MTS PO lines to stock buffers.

OCA-git-bot committed:
ddmrp 13.0.1.2.4

LoisRForgeFlow committed:
[13.0][ADD] ddmrp_chatter

oca-travis committed:
[UPD] Update ddmrp.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: ddmrp-13.0/ddmrp-13.0-ddmrp
Translate-URL: https://translation.odoo-community.org/projects/ddmrp-13-0/ddmrp-13-0-ddmrp/

JordiBForgeFlow committed:
[FIX] ddmrp: add 'partially_available' as a valid status when searching stock moves

OCA-git-bot committed:
ddmrp 13.0.1.2.5

LoisRForgeFlow committed:
[13.0][FIX] ddmrp: variables get overriden in each iteration.

OCA-git-bot committed:
ddmrp 13.0.1.3.0

LoisRForgeFlow committed:
[13.0][IMP] ddmrp: show a warning icon when a buffer has long term
supply, i.e. supply not included in the incoming qty used for the
NFP calculation.

LoisRForgeFlow committed:
[FIX] ddmrp: calculate execuction priority after finding a buffer link

LoisRForgeFlow committed:
[13.0][IMP] ddmrp: tests improvements:
* confirm picking on the create picking test helper methods and
  correct errors unveiled.
* add tests for incoming qty.

oca-travis committed:
[UPD] Update ddmrp.pot

OCA-git-bot committed:
ddmrp 13.0.1.4.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: ddmrp-13.0/ddmrp-13.0-ddmrp
Translate-URL: https://translation.odoo-community.org/projects/ddmrp-13-0/ddmrp-13-0-ddmrp/

Commit:
Add ddmrp_cron_actions_as_job

It makes calls to "cron_actions" run in queue jobs.

The jobs have an identity key with "identity_exact", meaning that only
one occurence of a job for the same buffer with the same arguments
(only_nfp) will be created at a time (e.g. when the state of a
stock.move is changed several times in the same transaction or in
a different transaction in a short timeframe).

It needs https://github.com/OCA/queue/pull/274 and
https://github.com/OCA/queue/pull/275

sebalix committed:
[IMP] ddmrp: call '_update_ddmrp_nfp' when creating moves

Commit:
Optimize StockBuffer.cron_actions()

Improvements in:
* _calc_qualified_demand
* _calc_incoming_dlt_qty

Before: ~1800ms
After: ~400ms

Tested on a database with 35'000 locations and 32'000 moves.

Details of the profiler below.

Rationale:

In these methods, "*_domain" methods search
all the locations and then filters moves based on their source/destination
locations. In `_calc_incoming_dlt_qty` the search is even done twice.
Searching for the locations first and for the moves with the location
(IN ids, NOT IN ids) domain is inefficient.
Filtering afterwards in memory (based on parent_path comparison) is much
faster and the recordset to filter should not grow too much as
restricted to a single product and a selection of "active" moves.

Also, in case there is no "active" move at all for a buffer, we spare
entirely the cost of searching for the locations.

Profiler details:

**Before**

calls     queries   ms
stock.buffer ------------------------ /odoo/external-src/ddmrp/ddmrp/models/stock_buffer.py, 1358

1         0         0.02          @profile
                                  def cron_actions(self, only_nfp=False):
                                      """This method is meant to be inherited by other modules in order to
                                      enhance extensibility."""
1         0         0.02              self.ensure_one()
1         0         0.0               if not only_nfp or only_nfp == "out":
1         24        844.56                self._calc_qualified_demand()
1         0         0.02              if not only_nfp or only_nfp == "in":
1         10        867.96                self._calc_incoming_dlt_qty()
1         6         65.18             self._calc_net_flow_position()
1         0         10.26             self._calc_planning_priority()
1         0         19.66             self._calc_execution_priority()
1         0         0.33              self.mrp_production_ids._calc_execution_priority()
1         1         3.85              self.mapped("purchase_line_ids")._calc_execution_priority()
1         0         0.01              if not only_nfp:
                                          # re-compoute red to force in cascade the recalculation of zones.
1         7         52.63                 self._compute_red_zone()
1         0         0.04              self.do_auto_procure()
1         0         0.0               return True

Total:
1         48        1864.54

**After**

calls     queries   ms
stock.buffer ------------------------ /odoo/external-src/ddmrp/ddmrp/models/stock_buffer.py, 1351

1         0         0.01          @profile
                                  def cron_actions(self, only_nfp=False):
                                      """This method is meant to be inherited by other modules in order to
                                      enhance extensibility."""
1         0         0.01              self.ensure_one()
1         0         0.0               if not only_nfp or only_nfp == "out":
1         18        162.8                 self._calc_qualified_demand()
1         0         0.01              if not only_nfp or only_nfp == "in":
1         8         63.24                 self._calc_incoming_dlt_qty()
1         9         77.79             self._calc_net_flow_position()
1         0         11.12             self._calc_planning_priority()
1         0         20.84             self._calc_execution_priority()
1         0         0.25              self.mrp_production_ids._calc_execution_priority()
1         1         4.1               self.mapped("purchase_line_ids")._calc_execution_priority()
1         0         0.01              if not only_nfp:
                                          # re-compoute red to force in cascade the recalculation of zones.
1         7         56.11                 self._compute_red_zone()
1         0         0.04              self.do_auto_procure()
1         0         0.01              return True

Total:
1         43        396.33

OCA-git-bot committed:
ddmrp 13.0.1.4.1

OCA-git-bot committed:
ddmrp 13.0.1.4.2

LoisRForgeFlow committed:
[13.0][FIX] ddmrp: do not consider internal reservations.

Operations done from locations within the buffer area/location
should not affect the on hand position of the buffer.

With this change, we rely on `stock.move.line`'s to compute the
qty reserved instead of using the `reserved_quantity` on the quant
which is the most accurate way to get the reserved qty. This also
eliminates `stock_available_unreserved` dependency.

oca-travis committed:
[UPD] Update ddmrp.pot

OCA-git-bot committed:
ddmrp 13.0.1.5.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: ddmrp-13.0/ddmrp-13.0-ddmrp
Translate-URL: https://translation.odoo-community.org/projects/ddmrp-13-0/ddmrp-13-0-ddmrp/

JordiBForgeFlow committed:
[IMP] Add imposed demand lead time to buffers.

When defined, the yellow zone will be computed as the ADU x Imposed Demand Lead Time, if the Imposed Demand Lead
Time is larger than the Replenishment Lead Time. This is particularily useful in situations of infrequent but
periodic demand. E.g. We receive a large order every 30 days, whereas the supplier takes 10 days to supply.

In this case the yellow zone must cover for the entire cycle of 30 days of demand.

In situations with infrequent demand the ADU tends to be very small, and every new order would be treated
as a spike, when in reality this is not an exceptional situation.

JordiBForgeFlow committed:
[IMP] create decimal precision 'Average Daily Usage'.

Before the decimal precision assigned to the ADU was the
Product Unit of Measure, but sometimes this precision can be set
to no digits, which would drive to an incorrect ADU calculation.

LoisRForgeFlow committed:
[13.0][IMP] ddmrp: archive buffers when archiving products.

OCA-git-bot committed:
ddmrp 13.0.1.5.1

OCA-git-bot committed:
ddmrp 13.0.1.6.0

oca-travis committed:
[UPD] Update ddmrp.pot

OCA-git-bot committed:
ddmrp 13.0.1.7.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: ddmrp-13.0/ddmrp-13.0-ddmrp
Translate-URL: https://translation.odoo-community.org/projects/ddmrp-13-0/ddmrp-13-0-ddmrp/

TDu committed:
[13.0][ADD] ddmrp, smart button from picking to buffer

Add a smart button on the stock.picking form to open the list of buffer
related to the product in the stock.picking.
By default it is filtered on Procurement Recommended.

jbaudoux committed:
ddmrp: when ddmrp_auto_update_nfp option is enabled, buffers should also be updated when move location src/dest is changed.

This fixes compatibility with modules like stock_dynamic_routing in OCA/wms that rewrites the location on pull/push rules.

sebalix committed:
[FIX] Do not include inventory loss in ADU computation

Inventory loss moves should not be included in the ADU past/future
computation (not relevant).

Commit:
Add option to limit replenishment to free quantity

This new feature targets buffers of type 'distributed'.

* Store the source location where the replenishment of a buffer should
  occur.
* The source location is taken from the default first route, so we
  assume the default route is a replenishment, otherwise the computation
  is skipped.
* When `stock_buffer_route` is installed, the route configured on the
  buffer is used to find the replenishment source location.
* A computed field shows the free quantity in the replenishment source
  location.
* When a buffer has no quantity in the replenishment source location, a
  new warning is displayed in the tree view.
* A new option on the buffer profile limits the quantity that can be
  replenished in the procurement wizard to the actual free quantity in
  the replenishment source location. The quantity we can procure for
  this profile will be min(recommended quantity, free quantity).

This is a huge win to avoid backorders for transfers created for
replenishments: we create transfers only for the quantity we have, and
if we do not have this quantity in practice, it reveals a stock issue.

LoisRForgeFlow committed:
[13.0][FIX] ddmrp: ADU fixed should use the ADU decimal precision

OCA-git-bot committed:
ddmrp 13.0.1.7.1

OCA-git-bot committed:
ddmrp 13.0.1.7.2

Commit:
Add reschedule procurement on distributed buffers

Add a configurable time for replenishment procurement on the profile.
This time is added to the planned datetime for the new moves, so they
are considered late when the replenishment time is overpassed.

To be able to do this, I had to change the "date_planned" field on
the procurement wizard from Date to Datetime.

When there is a calendar, takes the first working hour of the day and
add the max procurement time to it.

hitrosol committed:
Added translation using Weblate (Indonesian)

hitrosol committed:
Translated using Weblate (Indonesian)

Currently translated at 7.2% (19 of 264 strings)

Translation: ddmrp-13.0/ddmrp-13.0-ddmrp
Translate-URL: https://translation.odoo-community.org/projects/ddmrp-13-0/ddmrp-13-0-ddmrp/id/

sebalix committed:
[IMP] Option to include scrap moves in ADU calculation

oca-travis committed:
[UPD] Update ddmrp.pot

OCA-git-bot committed:
ddmrp 13.0.1.8.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: ddmrp-13.0/ddmrp-13.0-ddmrp
Translate-URL: https://translation.odoo-community.org/projects/ddmrp-13-0/ddmrp-13-0-ddmrp/

Commit:
Fix error in auto_procure when < free qty

When using the option to limit the procurement to free quantity, and the
free quantity is zero, the auto procurement tries to create procurement
for zero and it raises an error.

oca-travis committed:
[UPD] Update ddmrp.pot

OCA-git-bot committed:
ddmrp 13.0.1.9.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: ddmrp-13.0/ddmrp-13.0-ddmrp
Translate-URL: https://translation.odoo-community.org/projects/ddmrp-13-0/ddmrp-13-0-ddmrp/

LoisRForgeFlow committed:
[13.0][FIX] ddmrp: explicitly set the default wizard view.

This ensures that the functionality of the group that can
edit procurement qty will always work.

Commit:
Ignore max proc time for manufactured

OCA-git-bot committed:
ddmrp 13.0.1.10.0

LoisRForgeFlow committed:
[13.0][FIX] ddmrp: error when clearing produc_id field.

Because compute fields are called before saving, it can
happen that there is no product_id define (user is removing
it in edit mode). In this situation an error raised, this
commit fixes it.

OCA-git-bot committed:
ddmrp 13.0.1.10.1

Commit:
Fix fetching issue on crons methods

* Call commit once per chunk instead of once per record: commit flushes
  and clears the env, forcing a fetch of all the records of the chunk
  after each line.
* Remove `buffer.refresh()` in cron_ddmrp(): this deprecated method
  calls `invalidate_cache()` without ids and field names, which forces
  a fetch of all the records of the chunk at each line. If it is
  required to invalidate something (but I could not find any
  explanation), we should invalidate only the ids and fields that needs
  invalidation (hopefully only non-stored computed fields). Or at the
  very least, do not fetch the 50 records of the browse at once but one
  by one (but that'd be a pity).

The duration of cron_ddmrp() (when generating jobs with
ddmrp_cron_actions_as_job) goes from 8 minutes to 35 seconds.

Commit:
Invalidate computed non-stored fields in cron_actions

Fix tests:

* test_22_procure_recommended
* test_23_buffer_zones_all

It is possible that the invalidation is only required because of the
tests and would not be required at runtime. But this has been proposed
in the past: https://github.com/OCA/ddmrp/pull/59

However, this commit tries to invalidate the less possible fields to
limit the computations and data to fetch again.

OCA-git-bot committed:
ddmrp 13.0.1.11.0

Commit:
ddmrp: add dependency on stock_helper

To remove the duplicate implementation of
StockLocation.is_sublocation_of()

oca-travis committed:
[UPD] Update ddmrp.pot

OCA-git-bot committed:
ddmrp 13.0.1.12.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: ddmrp-13.0/ddmrp-13.0-ddmrp
Translate-URL: https://translation.odoo-community.org/projects/ddmrp-13-0/ddmrp-13-0-ddmrp/

LoisRForgeFlow committed:
[13.0][UPD] ddmrp: include ddmrp_warning in settings wizard

oca-travis committed:
[UPD] Update ddmrp.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: ddmrp-13.0/ddmrp-13.0-ddmrp
Translate-URL: https://translation.odoo-community.org/projects/ddmrp-13-0/ddmrp-13-0-ddmrp/

DavidBForgeFlow committed:
[IMP] ddmrp: black, isort, prettier

DavidBForgeFlow committed:
[MIG] ddmrp: Migration to 14.0

LoisRForgeFlow committed:
[13.0][FIX] ddmrp: main supplier recomputed without respecting the sequence.

When the current transaction changed the sequence, it may happen
that the sellers' recordset is not correctly sorted by default.
Doing it explicitly solves the problem.

TDu committed:
[13.0][FIX] ddmrp stock.buffer sellers

This is to be closer to how it works in Odoo standard.

So supplier wich are specific to the variant or not are
taken into account.

LoisRForgeFlow committed:
[13.0][FIX] ddmrp: incorrect default ordering.

Default prioritization in DDMRP is color first, NFP percentage(!)
second.

LoisRForgeFlow committed:
[13.0][IMP] ddmrp: add ddmrp_qty_multiple_tolerance option.

Set a tolerance value to apply to quantity multiple in stock
buffers. If the quantity needed is below this tolerance
threshold, the recommended quantity will be reduced a bit
instead of adding another bucket. The value is a percentage
of the quantity multiple.

oca-travis committed:
[UPD] Update ddmrp.pot

OCA-git-bot committed:
[UPD] README.rst

jbaudoux committed:
Fix console warnings due to list view renderer

The error "Error while parsing JSON SyntaxError: Unexpected token ' in
JSON at position 1" was logged in the console when for example the field
contains a {'no_create': True}. This is not a JSON valid string.
Changed to use the same parsing as in odoo basic_view.

oca-travis committed:
[UPD] Update ddmrp.pot

OCA-git-bot committed:
ddmrp 14.0.1.1.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: ddmrp-14.0/ddmrp-14.0-ddmrp
Translate-URL: https://translation.odoo-community.org/projects/ddmrp-14-0/ddmrp-14-0-ddmrp/

OCA-git-bot committed:
ddmrp 14.0.1.1.1

LoisRForgeFlow committed:
[FIX] ddmrp: do not show distributed_reschedule_max_proc_time in
profile name if not in use.

mmequignon committed:
Fix watched fields for main supplier compute on stock buffer

mmequignon committed:
Rely on the default odoo sort to compute default supplier

oca-travis committed:
[UPD] Update ddmrp.pot

OCA-git-bot committed:
ddmrp 14.0.1.2.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: ddmrp-14.0/ddmrp-14.0-ddmrp
Translate-URL: https://translation.odoo-community.org/projects/ddmrp-14-0/ddmrp-14-0-ddmrp/

LoisRForgeFlow committed:
[FIX] ddmrp: since date planned is now a datetime, the limit to
consider incoming moves should also be a datetime and use the
same logic.

Otherwise, a just-created incoming supply could be ignored
when rescheduled to the next day (in UTC, the problem is present
in all timezones but with an offset). This commit includes a
test case to highlight the issue.

LoisRForgeFlow committed:
[FIX] ddmrp: extra lead time should only affect sizing.

Extra lead time is only a sizing factor, the description is
updated to make it clearer and the computation of dlt and
zones adapted.

oca-travis committed:
[UPD] Update ddmrp.pot

OCA-git-bot committed:
ddmrp 14.0.1.2.1

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: ddmrp-14.0/ddmrp-14.0-ddmrp
Translate-URL: https://translation.odoo-community.org/projects/ddmrp-14-0/ddmrp-14-0-ddmrp/

OCA-git-bot committed:
ddmrp 14.0.1.2.2

TDu committed:
[13.0][IMP] ddmrp improve po line on hand status

Add the quantity ordered
Allow multiple edit
Allow editing scheduled date

oca-travis committed:
[UPD] Update ddmrp.pot

OCA-git-bot committed:
ddmrp 14.0.1.3.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: ddmrp-14.0/ddmrp-14.0-ddmrp
Translate-URL: https://translation.odoo-community.org/projects/ddmrp-14-0/ddmrp-14-0-ddmrp/

jbaudoux committed:
Fix security on make procurement on stock buffer

LoisRForgeFlow committed:
[FIX] ddmrp: ddmrp_comment should be editable in PO line.

This has probably been missing in migration to v13.

LoisRForgeFlow committed:
[14.0][UPD] ddmrp: adapt chart to new bokeh version 2.3.1.

The library has been updated in the depency `web_widget_bokeh_chart`
at OCA/web.

OCA-git-bot committed:
ddmrp 14.0.1.3.1

OCA-git-bot committed:
ddmrp 14.0.1.4.0

TDu committed:
[13.0][IMP] ddmrp improve po line on hand status

Remove the restricting domain showing only lines with a buffer. And
replace it with a default filter.

Finally remove the widget date which is not timezone friendly

LoisRForgeFlow committed:
[13.0][IMP] ddmrp: add hook to extend auto-update buffers.

oca-travis committed:
[UPD] Update ddmrp.pot

OCA-git-bot committed:
ddmrp 14.0.1.5.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: ddmrp-14.0/ddmrp-14.0-ddmrp
Translate-URL: https://translation.odoo-community.org/projects/ddmrp-14-0/ddmrp-14-0-ddmrp/

LoisRForgeFlow committed:
[14.0][FIX] ddmrp: buffer profiles must be no-update

Any change in the profiles is overridden when updating the module.
Also this trigger the recomputation of the buffer zones, as
they depend on buffer profile changes, this is unwanted.

CSV data does not support no-update tag so the date is converted to
xml.

TDu committed:
[13.0][IMP] ddmrp add used in MRP smart button

Add a smart button at the top of the buffer form to open the list of BOM,
same than on the product form.

OCA-git-bot committed:
ddmrp 14.0.1.6.1

TDu committed:
[14.0][IMP] ddmrp add product vendor code

Add the product vendor code on the stock buffer based on the already
computed main supplier.

Forward port from 13.0
* https://github.com/OCA/ddmrp/pull/148

TDu committed:
[IMP] ddmrp: Add yearly consumption button stock.buffer

atchuthan committed:
[FIX] ddmrp: parameter not passed properly

OCA-git-bot committed:
ddmrp 14.0.1.6.2

LoisRForgeFlow committed:
[14.0][MIG] ddmrp: fix refresh buffer button string

oca-travis committed:
[UPD] Update ddmrp.pot

OCA-git-bot committed:
ddmrp 14.0.1.7.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: ddmrp-14.0/ddmrp-14.0-ddmrp
Translate-URL: https://translation.odoo-community.org/projects/ddmrp-14-0/ddmrp-14-0-ddmrp/

oca-travis committed:
[UPD] Update ddmrp.pot

OCA-git-bot committed:
ddmrp 14.0.1.8.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: ddmrp-14.0/ddmrp-14.0-ddmrp
Translate-URL: https://translation.odoo-community.org/projects/ddmrp-14-0/ddmrp-14-0-ddmrp/

mmequignon committed:
[IMP] ddmrp: Ease filtering and usage of buffer tree view

sebalix committed:
[IMP] ddmrp: add an index on 'stock_move_line.state'

This allows to improve the performance when accessing the list of DDMRP
buffers.
The field `_compute_product_available_qty` method of `stock.buffer` is
searching for `stock.move.line` records based on their status.

LoisRForgeFlow committed:
[FIX] ddmrp: reuse `_past_moves_domain` in yearly consuption action

oca-travis committed:
[UPD] Update ddmrp.pot

OCA-git-bot committed:
ddmrp 14.0.1.9.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: ddmrp-14.0/ddmrp-14.0-ddmrp
Translate-URL: https://translation.odoo-community.org/projects/ddmrp-14-0/ddmrp-14-0-ddmrp/

oca-travis committed:
[UPD] Update ddmrp.pot

OCA-git-bot committed:
ddmrp 14.0.1.10.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: ddmrp-14.0/ddmrp-14.0-ddmrp
Translate-URL: https://translation.odoo-community.org/projects/ddmrp-14-0/ddmrp-14-0-ddmrp/

kv1612 committed:
[MIG] ddmrp_product_replace: Migration to 14.0

oca-travis committed:
[UPD] Update ddmrp.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: ddmrp-14.0/ddmrp-14.0-ddmrp
Translate-URL: https://translation.odoo-community.org/projects/ddmrp-14-0/ddmrp-14-0-ddmrp/

alexis-via committed:
[FIX] ddmrp: fix crash in BOM structure report

OCA-git-bot committed:
ddmrp 14.0.1.10.1

sebalix committed:
[IMP] ddmrp: select sellers depending on dates

Select the sellers like in standard Odoo, based on 'date_start' and
'date_end' fields.

TDu committed:
Disable unarchiving buffer on product activation

Having the stock buffer automatically unarchived when the state of a
product changes is dangerous because it can lead to product being
ordered again when it should not anymore.

The issue encountered seemed to be related to how Odoo handles product
variant. And some changes in the product attribute line of a product
would call a write on the active field of related products variant.

An improvement that could be added (maybe coming soon) is to have on
the product form the information of existing archived stock buffers.

Commit:
[IMP] ddmrp: fix test with working time for plan_days

oca-travis committed:
[UPD] Update ddmrp.pot

OCA-git-bot committed:
ddmrp 14.0.1.11.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: ddmrp-14.0/ddmrp-14.0-ddmrp
Translate-URL: https://translation.odoo-community.org/projects/ddmrp-14-0/ddmrp-14-0-ddmrp/

OCA-git-bot committed:
ddmrp 14.0.1.11.1

TDu committed:
Fix quantity_in_progress computation

If a purchase order line is related to multiple buffers but not all of
those buffer are included in `self` then the computation crash.

i-vyshnevska committed:
[FIX] ddmrp: remove direct read of client action

cormaza committed:
[IMP] ddmrp: black, isort, prettier

ChrisOForgeFlow committed:
[15.0][MIG] ddmrp

sebalix committed:
ddmrp: add index on 'distributed_source_location_id'

The filter on 'distributed_source_location_qty' is using a
search method querying the 'distributed_source_location_id' field having
no index.

Commit:
[UPD] Update ddmrp.pot

OCA-git-bot committed:
[UPD] README.rst

i-vyshnevska committed:
[FIX] ddmrp: method moved to purchase.order.line model

OCA-git-bot committed:
ddmrp 15.0.1.0.1

OCA-git-bot committed:
ddmrp 15.0.1.0.2

OCA-git-bot committed:
ddmrp 15.0.1.0.3

Commit:
[UPD] Update ddmrp.pot

OCA-git-bot committed:
ddmrp 15.0.1.1.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: ddmrp-15.0/ddmrp-15.0-ddmrp
Translate-URL: https://translation.odoo-community.org/projects/ddmrp-15-0/ddmrp-15-0-ddmrp/

OCA-git-bot committed:
ddmrp 15.0.1.2.0

OCA-git-bot committed:
ddmrp 15.0.1.2.1

BernatPForgeFlow committed:
[FIX] ddmrp: remove mrp_move.py as it is not used

Commit:
[UPD] Update ddmrp.pot

OCA-git-bot committed:
ddmrp 15.0.1.2.2

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: ddmrp-15.0/ddmrp-15.0-ddmrp
Translate-URL: https://translation.odoo-community.org/projects/ddmrp-15-0/ddmrp-15-0-ddmrp/

JordiBForgeFlow committed:
[FIX] ddmrp: adu field needs to be explicity written when adu_fixed is set
The onchange only has a visual effect. Actual changes to adu must be made in
the create and write methods.

OCA-git-bot committed:
ddmrp 15.0.1.2.3

BernatPForgeFlow committed:
[FIX] ddmrp: get_stock_move_values in stock_rule

OCA-git-bot committed:
ddmrp 15.0.1.2.4

sebalix committed:
[IMP] ddmrp: add index on 'stock_move.created_purchase_line_id'

Commit:
[UPD] Update ddmrp.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: ddmrp-15.0/ddmrp-15.0-ddmrp
Translate-URL: https://translation.odoo-community.org/projects/ddmrp-15-0/ddmrp-15-0-ddmrp/

LoisRForgeFlow committed:
[FIX] ddmrp: clean warning in logs

remove unneded attributes on related field definition.

Commit:
[UPD] Update ddmrp.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: ddmrp-15.0/ddmrp-15.0-ddmrp
Translate-URL: https://translation.odoo-community.org/projects/ddmrp-15-0/ddmrp-15-0-ddmrp/

Commit:
[UPD] Update ddmrp.pot

OCA-git-bot committed:
ddmrp 15.0.1.3.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: ddmrp-15.0/ddmrp-15.0-ddmrp
Translate-URL: https://translation.odoo-community.org/projects/ddmrp-15-0/ddmrp-15-0-ddmrp/

ChrisOForgeFlow committed:
[15.0][FIX] ddmrp, case when origin warehouse has 2 steps to delivery

JordiBForgeFlow committed:
[IMP] ddmrp: add hooks in make procurement wizard for extensibility

OCA-git-bot committed:
ddmrp 15.0.1.3.1

ibuioli committed:
Added translation using Weblate (Spanish (Argentina))

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (291 of 291 strings)

Translation: ddmrp-15.0/ddmrp-15.0-ddmrp
Translate-URL: https://translation.odoo-community.org/projects/ddmrp-15-0/ddmrp-15-0-ddmrp/es_AR/

BernatPForgeFlow committed:
[IMP] ddmrp: Add hook to get produce delay

A hook is added to get production delay because subcontracting BoMs calculate production time differently, from the time of the first vendor with subcontracted type.

OCA-git-bot committed:
ddmrp 15.0.1.3.2

BernatPForgeFlow committed:
[FIX] ddmrp_*: Adapt call to current thread

BernatPForgeFlow committed:
[IMP] ddmrp: Add links

- Add link to Stock Demand Estimates from Stock Buffer
- Add link to MRP BoM from Stock Buffer
- Add link to Stock Buffers from Product Variants
- Add link to Stock Buffer from MRP BoM Line

OCA-git-bot committed:
ddmrp 15.0.1.3.3

BernatPForgeFlow committed:
[IMP] ddmrp: Redesign stock buffer form view

- On the top, we will have contextualization about the bufffer.
- On the left side, we will have settings and info about the configuration of the buffer and procurements.
- On the right side, we will have information about the buffer.

Apart from this, you will be able to access the source of ADU, Qualified demand and Incoming (DLT) calculations.

Commit:
[UPD] Update ddmrp.pot

OCA-git-bot committed:
ddmrp 15.0.1.5.0

Commit:
[UPD] Update ddmrp.pot

OCA-git-bot committed:
ddmrp 15.0.1.6.0

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: ddmrp-15.0/ddmrp-15.0-ddmrp
Translate-URL: https://translation.odoo-community.org/projects/ddmrp-15-0/ddmrp-15-0-ddmrp/

LoisRForgeFlow committed:
[IMP] ddmrp: introduce specific security groups

Try to give a bit more of flexibility on who can modify ddmrp
master data (new ddmrp manager group) and stock buffers (new stock
stock buffer maintainer group).

LoisRForgeFlow committed:
[IMP] ddmrp: add execution chart to buffer summary

LoisRForgeFlow committed:
[IMP] ddmrp_history: use helper method to get colors

BernatPForgeFlow committed:
[FIX] ddmrp: Remove active in domains

OCA-git-bot committed:
ddmrp 15.0.1.6.1

LoisRForgeFlow committed:
[FIX] ddmrp: remove unneded inheritance

upstream _generate_finished_moves does not exist since v14,
the equivalent extesion for should be done in
_get_move_finished_values, however is not needed as Odoo has
already corrected the issue.

Commit:
[UPD] Update ddmrp.pot

OCA-git-bot committed:
ddmrp 15.0.1.7.0

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: ddmrp-15.0/ddmrp-15.0-ddmrp
Translate-URL: https://translation.odoo-community.org/projects/ddmrp-15-0/ddmrp-15-0-ddmrp/

Commit:
[UPD] Update ddmrp.pot

OCA-git-bot committed:
ddmrp 15.0.1.8.0

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: ddmrp-15.0/ddmrp-15.0-ddmrp
Translate-URL: https://translation.odoo-community.org/projects/ddmrp-15-0/ddmrp-15-0-ddmrp/

BernatPForgeFlow committed:
[IMP] ddmrp: Add hook to get ADU calculation horizon

BernatPForgeFlow committed:
[IMP] ddmrp: Add missing ADU calculation horizon calls

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (314 of 314 strings)

Translation: ddmrp-15.0/ddmrp-15.0-ddmrp
Translate-URL: https://translation.odoo-community.org/projects/ddmrp-15-0/ddmrp-15-0-ddmrp/es_AR/

OCA-git-bot committed:
ddmrp 15.0.1.8.1

OCA-git-bot committed:
ddmrp 15.0.1.8.2

BernatPForgeFlow committed:
[MIG] ddmrp: Migration to 16.0

[IMP] ddmrp: Add OWL functionality

[FIX] ddmrp: fix migration changes

JordiBForgeFlow committed:
[IMP] ddmrp: add option to install ddmrp_purchase_hide_onhand_status

JordiBForgeFlow committed:
[FIX] ddmrp: use warehouse_id in location instead of method get_warehouse

BernatPForgeFlow committed:
[IMP] ddmrp: Add BoM report functionalities

LoisRForgeFlow reviewed (approved):
Functional and technical review, generally is working good. There are small glitches in the buffer list view and bokeh charts but I think it is better to merge this first version to unlock the rest of migrations depending on this one. Those small issues will be fixed in new PRs.

LoisRForgeFlow commented:
/ocabot merge nobump

LoisRForgeFlow committed:
[FIX] ddmrp: buffer_a in test should use a manufactured buffer profile

all test adjusted accordingly and added new test for lead_days as
it is useless in manufactured buffer profiles

LoisRForgeFlow commented:
Issues in tests fixed, trying to merge again:

/ocabot merge nobump
